### PR TITLE
[AP-121]-Friend req count

### PR DESCRIPTION
**Pull request by KemoPaw**
**What this PR does** 📖

https://satellite-im.atlassian.net/browse/AP-121

**Which issue(s) this PR fixes** 🔨

Fixes #

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Review by stavares843 — CHANGES_REQUESTED** [review]
neat 🎉 

any suggestions on how to avoid 2 digits to be overlapping the copy of the request? maybe 9+ instead? we can ask on the channel as well

Captura de ecrã 2021-12-07, às 23 41 39

**Comment by RetroPronghorn** [comment]
We should add some padding right to the button when there are notifications to avoid the overlap

File: components/tailored/friends/nav/Nav.less [hunk 1]
```diff
@@ -10,7 +10,24 @@
...
+    top: 0.25rem;
+    right: 0.25rem;
+    font-size: @micro-text-size;
+    background-color: @primary-color;
+    border-radius: 2px;
+    line-height: 1.2;
+    padding: 0.15rem 0.5rem;
+    color: @text;
+    font-family: @primary-font;
+  }
```
Comment (josephmcg) [inline components/tailored/friends/nav/Nav.less:28]: avoid using important. probably just switch the order of these

File: components/tailored/friends/nav/Nav.html [hunk 2]
```diff
@@ -1,34 +1,44 @@
 <div id="friends-nav">
   <div v-if="$device.isMobile" class="spacer"></div>
-  <InteractablesButton 
-    size="small" 
-    :text="`${$device.isMobile ? '' : 'Active'}`" 
+  <InteractablesButton
+    size="small"
+    :text="`${$device.isMobile ? '' : 'Active'}`"
```
Comment (josephmcg) [inline components/tailored/friends/nav/Nav.html:5]: prob don't need template literal

File: components/tailored/friends/nav/Nav.html [hunk 3]
```diff
@@ -1,34 +1,44 @@
...
-    :text="`${$device.isMobile ? '' : 'Requests'}`" 
-    :type="route === 'requests' ? 'primary' : 'dark'"
-    :action="() => setRoute('requests')"
+  <div>
+    <InteractablesButton
+      size="small"
+      :text="`${$device.isMobile ? '' : 'Requests'}`"
+      :type="route === 'requests' ? 'primary' : 'dark'"
+      :action="() => setRoute('requests')"
+      :class="this.friends.incomingRequests.length ? 'big-button' : 'button' "
```
Comment (josephmcg) [inline components/tailored/friends/nav/Nav.html:17]: same here^

File: components/tailored/friends/nav/Nav.html [hunk 4]
```diff
@@ -1,34 +1,44 @@
...
+    </InteractablesButton>
+  </div>
+  <InteractablesButton
     size="small"
-    :text="`${$device.isMobile ? '' : 'Blocked'}`" 
+    :text="`${$device.isMobile ? '' : 'Blocked'}`"
     :type="route === 'blocked' ? 'primary' : 'dark'"
-    :action="() => setRoute('blocked')">
+    :action="() => setRoute('blocked')"
+  >
```
Comment (josephmcg) [inline components/tailored/friends/nav/Nav.html:33]: and here^

File: components/tailored/friends/nav/Nav.html [hunk 5]
```diff
@@ -1,34 +1,44 @@
...
   </InteractablesButton>
-  <InteractablesButton 
+  <InteractablesButton
     size="small"
-    :text="`${$device.isMobile ? 'Add' : 'Add Friend'}`" 
+    :text="`${$device.isMobile ? 'Add' : 'Add Friend'}`"
     :type="route === 'add' ? 'primary' : 'dark'"
-    :action="() => setRoute('add')">
+    :action="() => setRoute('add')"
+  >
```
Comment (josephmcg) [inline components/tailored/friends/nav/Nav.html:41]: and here

File: components/tailored/friends/nav/Nav.vue [hunk 6]
```diff
@@ -16,6 +17,20 @@ export default Vue.extend({
     UserPlusIcon,
     UserXIcon,
   },
+  computed: {
```
Comment (josephmcg) [inline components/tailored/friends/nav/Nav.vue:20]: switch order of props and computed to make eslint happy

File: components/tailored/friends/nav/Nav.vue [hunk 7]
```diff
@@ -16,6 +17,20 @@ export default Vue.extend({
...
+    ...mapState(['friends']),
+    /**
+     * makeFriendReqTag: generates the Request tag number on the Request Button
+     * depending on the number of ongoing friend requests, it will generate a different replyLength
+     */
+    makeFriendReqTag() {
+      const replyLength = this.friends.incomingRequests.length
+      if (replyLength > 100) {
+        return '99+'
+      }
```
Comment (josephmcg) [inline components/tailored/friends/nav/Nav.vue:30]: if (replyLength >**=** 100) {

otherwise it would show 100

**Review by josephmcg — CHANGES_REQUESTED** [review]

File: components/tailored/friends/nav/Nav.html (hunk 2 above)
Comment (KemoPaw) [reply components/tailored/friends/nav/Nav.html:5]: Okay, will change that one!

File: components/tailored/friends/nav/Nav.less (hunk 1 above)
Comment (KemoPaw) [reply components/tailored/friends/nav/Nav.less:28]: @josephmcg Okay, will change & test this out.

File: components/tailored/friends/nav/Nav.vue (hunk 6 above)
Comment (KemoPaw) [reply components/tailored/friends/nav/Nav.vue:20]: @josephmcg Okay, thank you very much for catching that one!

File: components/tailored/friends/nav/Nav.vue (hunk 7 above)
Comment (KemoPaw) [reply components/tailored/friends/nav/Nav.vue:30]: @josephmcg Good catch, was testing out something else beforehand and missed this one. Much appreciated!

**Review by WanderingHogan — APPROVED** [review]

File: components/tailored/friends/nav/Nav.less [hunk 8]
```diff
@@ -6,11 +6,27 @@
   &:nth-last-child(.button) {
     margin-right: 0;
   }
+  .big-button {
+    justify-self: flex-end;
+  }
```
Comment (josephmcg) [inline components/tailored/friends/nav/Nav.less:11]: this can be removed

**Review by RetroPronghorn — APPROVED** [review]

File: components/tailored/friends/nav/Nav.vue [hunk 9]
```diff
@@ -26,6 +27,37 @@ export default Vue.extend({
...
   },
+  computed: {
+    ...mapState(['friends']),
+    /**
+     * friendRequestPadding: calculates how much padding to add to a button depending on friendRequest length
+     * depending on the number of ongoing friend requests, it will generate a different px amount for padding-right
+     */
+    friendRequestPadding() {
+      const currentIncomingRequests = this.friends.incomingRequests.length.toString()
+      const length = this.friends.incomingRequests.length.toString().length
```
Comment (josephmcg) [inline components/tailored/friends/nav/Nav.vue:38]: could sub for
const length = currentIncomingRequests.length

**Review by stavares843 — APPROVED** [review]

File: components/tailored/friends/nav/Nav.html [hunk 10]
```diff
@@ -1,34 +1,45 @@
...
-    :text="`${$device.isMobile ? '' : 'Requests'}`" 
-    :type="route === 'requests' ? 'primary' : 'dark'"
-    :action="() => setRoute('requests')"
+  <div>
+    <InteractablesButton
+      size="small"
+      :text="$device.isMobile ? '' : 'Requests'"
+      :type="route === 'requests' ? 'primary' : 'dark'"
+      :action="() => setRoute('requests')"
+      :class="friends.incomingRequests.length ? 'big-button' : 'button' "
```
Comment (josephmcg) [inline components/tailored/friends/nav/Nav.html:17]: this line can be removed. class is set in the component

File: components/tailored/friends/nav/Nav.less [hunk 11]
```diff
@@ -6,11 +6,27 @@
...
   .button {
     margin: 0 (@light-spacing / 2);
     justify-self: flex-end;
   }
+  .label {
+    position: absolute;
+    top: 0.25rem;
+    right: 0.25rem;
```
Comment (josephmcg) [inline components/tailored/friends/nav/Nav.less:19]: there's a global for this
@xlight-spacing

**Review by josephmcg — COMMENTED** [review]
a couple minor things, mostly removing dead code

File: components/tailored/friends/nav/Nav.html (hunk 10 above)
Comment (josephmcg) [reply components/tailored/friends/nav/Nav.html:17]: *class line. highlight was wonky

**Comment by KemoPaw** [comment]
@josephmcg Changes have been made, thank you for the notes/feedback!

**Review by josephmcg — APPROVED** [review]
🚀

**Pull request merged**